File: 1/scala/1.scala
import euler._

object Main extends App {

  type Set = Int => Boolean

  val maximum = 999

  def reduce(s: Set, res: Int = 0, cur: Int = 0): Int =
    if (cur > maximum) res
    else if (s(cur)) reduce(s, res + cur, cur + 1)
    else reduce(s, res, cur + 1)

  def set(x: Int): Boolean =
    x % 3 == 0 || x % 5 == 0

  println(reduce(set))

}

File: 1/python/1.py
import euler

def multiple_of_three(x): return x % 3 == 0
def multiple_of_five(x):  return x % 5 == 0

def multiple_of_three_or_five(x): return multiple_of_three(x) or multiple_of_five(x)

multiples_of_three_or_five = filter(multiple_of_three_or_five, range(1, 1000))

answer = reduce(euler.sum, multiples_of_three_or_five)

print(answer)
